### PR TITLE
Fix detection of org-scoped packages

### PR DESCRIPTION
**Pull request by lukehoban**
Alternative fix for #260, adds tests and makes a more targeted fix than #261 which appears to introduce other issues.

Use path APIs instead of regexp in workaround for incorrect package names coming from `read-package-tree` dependency when seeing organization-scoped packages.

Add tests for additional cases of organization-scoped packages (at the root as well as nested).

### CODE REVIEW

File: overlays/nodejs/serverless/function.ts [hunk 1]
```diff
@@ -309,17 +311,17 @@ function addPackageAndDependenciesToSet(s: Set<string>, root: Package, pkg: stri
...
 function findDependency(root: Package, name: string) {
-    for(;root;root = root.parent) {
+    for (; root; root = root.parent) {
         for (var child of root.children) {
             let childName = child.name;
             // Note: `read-package-tree` returns incorrect `.name` properties for packages in an orgnaization - like
             // `@types/express` or `@protobufjs/path`.  Compute the correct name from the `path` property instead.
             // Match any name that ends with something that looks like `@foo/bar`.
-            const match = /\/\@[^\/]*\/[^\/]*$/.exec(child.path);
+            const match = /\@[^\/]*\/[^\/]*$/.exec(child.path);
```
Comment (joeduffy) [inline overlays/nodejs/serverless/function.ts]: I personally vote against using regexes to match paths like this.  I'd prefer to see us using real filesystem operations that parse the path more precisely.  (For example, this regex will now match `some@foo/bar`, which I'm not sure is intentional?  And if it is, I'd rather state this more precisely than just "happening" to match it using a regex.)

Comment (lukehoban) [reply overlays/nodejs/serverless/function.ts]: Yes - we’ll move this over to real path operations next. Just opened this to verify that things were working correctly with the minimal change.

**Comment by joeduffy** [comment]
Great to have a test (thought it probably makes sense to pull it out of `examples/` and into `tests/` -- realize this is a work in progress).  I closed out #261.

I assume you didn't see the other issue I saw with a missing `loglevel` library?  (Since the tests passed, I assume not ... however, it could just be that a different set of packages lead to a different layout.)

Do you think we can land today?  I was hoping to focus on this stuff with my blog post, although worst case I can workaround the issue with some package renaming (or just not use Slack).

**Comment by joeduffy** [comment]
> Do you think we can land today?

By "land", I meant commit and cut a new release of `@pulumi/aws`.

If not today, tomorrow is fine also.

**Comment by lukehoban** [comment]
> I assume you didn't see the other issue I saw with a missing loglevel library? (Since the tests passed, I assume not ... however, it could just be that a different set of packages lead to a different layout.)

Yeah - per

**Comment by joeduffy** [comment]
Awesome 🎉

**Pull request merged**